inc/class-addon-manager.php

```diff
@@ -0,0 +1,263 @@
+	 *
+	 * @param stdClass $subscription The subscription to convert.
+	 *
+	 * @return stdClass The converted subscription.
+	 */
+	protected function convert_subscription_to_plugin( $subscription ) {
+		return (object) array(
+			'new_version'   => $subscription->product->version,
+			'name'          => $subscription->product->name,
+			'slug'          => $subscription->product->slug,
+			'url'           => $subscription->product->url,
+			'last_update'   => $subscription->product->lastUpdated,
+			'homepage'      => $subscription->product->storeUrl,
+			'download_link' => $subscription->product->download,
+			'sections'      =>
+				array(
+					'changelog' => $subscription->product->changelog,
+				),
+		);
+	}
+
+	/**
+	 * Checks if the given plugin_file belongs to a Yoast addon.
+	 *
+	 * @param string $plugin_file Path to the plugin.
+	 *
+	 * @return bool True when plugin file is for a Yoast addon.
+	 */
+	protected function is_yoast_addon( $plugin_file ) {
+		return $this->get_slug_by_plugin_file( $plugin_file ) !== '';
+	}
+
+	/**
+	 * Retrieves the addon slug by given plugin file path.
+	 *
+	 * @param string $plugin_file The file path to the plugin.
+	 *
+	 * @return string The slug when found or empty string when not.
+	 */
+	protected function get_slug_by_plugin_file( $plugin_file ) {
+		foreach ( self::$addons as $addon => $addon_slug ) {
+			if ( strpos( $plugin_file, $addon ) !== false ) {
+				return $addon_slug;
+			}
+		}
+
+		return '';
+	}
+
+	/**
+	 * Retrieves the installed Yoast addons.
+	 *
+	 * @return array The installed plugins.
+	 */
+	protected function get_installed_addons() {
+		$plugins      = $this->get_plugins();
+		$plugin_files = array_filter( array_keys( $plugins ), array( $this, 'is_yoast_addon' ) );
+
+		$installed_plugins = array();
+		foreach ( $plugin_files as $plugin_file ) {
+			$installed_plugins[ $plugin_file ] = $plugins[ $plugin_file ];
+		}
+
+		return $installed_plugins;
+	}
+
+
+	/**
+	 * Retrieves the current sites from the API.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return bool|stdClass Object when request is successful. False if not.
+	 */
+	protected function request_current_sites() {
+		$api_request = new WPSEO_MyYoast_Api_Request( 'sites/current' );
+		if ( $api_request->fire() ) {
+			return $api_request->get_response();
+		}
+
+		return false;
+	}
+
+	/**
+	 * Retrieves the transient value with the site information.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return stdClass|false The transient value.
+	 */
+	protected function get_site_information_transient() {
+		return get_transient( 'wpseo_site_information' );
+	}
+
+	/**
+	 * Sets the site information transient.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @param stdClass $site_information The site information to save.
+	 *
+	 * @return void
+	 */
+	protected function set_site_information_transient( $site_information ) {
+		set_transient( 'wpseo_site_information', $site_information, DAY_IN_SECONDS );
```

Comment:
I would set the transient key in a `const` so it can be used externally.

It's likely that we'll have future actions where we want to clear this transient ( for example: activating/deactivating an extension, subscription-related notifications from MyYoast etc. ).